var/spack/repos/builtin/packages/openldap/package.py

```diff
@@ -0,0 +1,87 @@
+# Copyright 2013-2019 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class Openldap(AutotoolsPackage):
+    """
+    OpenLDAP Software is an open source implementation of the Lightweight
+    Directory Access Protocol. The suite includes:
+
+    slapd - stand-alone LDAP daemon (server)
+    libraries implementing the LDAP protocol, and
+    utilities, tools, and sample clients.
+    """
+
+    homepage = "https://www.openldap.org/"
+    url      = "ftp://ftp.openldap.org/pub/OpenLDAP/openldap-release/openldap-2.4.9.tgz"
+
+    version('2.4.48', sha256='d9523ffcab5cd14b709fcf3cb4d04e8bc76bb8970113255f372bc74954c6074d')
+
+    variant('client_only', default=True, description='Client only installation')
+    variant('icu', default=False, description='Build with unicode support')
+    variant('openssl', default=False, description='Use OpenSSL for TLS support')
```

Comment:
I would use a multi-valued variant where you can choose between `openssl` and `gnutls`.

Reply:
I am wondering I can make variant default to 'none' if +client_only or else default to 'gnutls'?
```python
    variant('tls', default='none', description='Build with TLS support',
            values=('gnutls', 'openssl', 'none'), multi=False, when='~client_only'
    )

```

Reply:
There's is currently no way to do that. The variant directive does not support a `when` argument.

Reply:
I guess I can make it default to 'gnutls' or openssl but then reset it if client_only like:
```
if `+client_only` in self.spec:
    variant('tls=none')
```
is there a simple way to accomplish this with `variant` or with `self.spec`

Reply:
Nope, unfortunately there's no way to do that.

Reply:
In that case, I can make tls default to 'gnutls' and add "with-tls=..." configure arg only if '~client_only' but then for a client_only installation tls variant will appear like tls=gnutls although gnutls isn't used in the build which is misleading:
```
$ spack find -lv openldap                                                                                
==> 1 installed package
-- linux-centos7-x86_64 / gcc@7.4.0 -----------------------------
ropnzob openldap@2.4.48+client_only~icu~perl tls=gnutls
```

Reply:
That seems reasonable to me.